tests/test_events.py

```diff
@@ -1,11 +1,59 @@
 
 
-def test_create_late_event(past_events_timestamps, setup, tollways):
-    late_event = create_late_event(
-        events_log=past_events_timestamps, fake=setup.get("faker"), tollways=tollways
+def test_create_late_event_seconds(late_events, setup, tollways):
+    late_event_ts = create_late_event(
+        events_log=late_events, fake=setup.get("faker"), tollways=tollways, time_unit="seconds"
+    )
+
+    late_event_ts = datetime.strptime(late_event_ts["timestamp"], TIMESTAMP_FORMAT)
+    for ts in late_events["late_events"]["seconds"]:
+        current_event_ts = datetime.strptime(ts, TIMESTAMP_FORMAT)
+        if late_event_ts.microsecond == current_event_ts.microsecond:
+            difference_in_seconds = abs(current_event_ts - late_event_ts)
+            assert TIME_UNIT["seconds"]["min"] <= difference_in_seconds.seconds <= TIME_UNIT["seconds"]["max"]
+
+
+def test_create_late_event_minutes(late_events, setup, tollways):
+    late_event_ts = create_late_event(
+        events_log=late_events, fake=setup.get("faker"), tollways=tollways, time_unit="minutes"
+    )
+
+    late_event_ts = datetime.strptime(late_event_ts["timestamp"], TIMESTAMP_FORMAT)
+    for ts in late_events["late_events"]["minutes"]:
+        current_event_ts = datetime.strptime(ts, TIMESTAMP_FORMAT)
+        if late_event_ts.microsecond == current_event_ts.microsecond:
+            difference_in_minutes = abs(current_event_ts - late_event_ts).seconds // 60
+            assert TIME_UNIT["minutes"]["min"] <= difference_in_minutes <= TIME_UNIT["minutes"]["max"]
+
+
+def test_create_late_event_hours(late_events, setup, tollways):
+    late_event_ts = create_late_event(
+        events_log=late_events, fake=setup.get("faker"), tollways=tollways, time_unit="hours"
     )
-    assert late_event.get("timestamp") in past_events_timestamps.get("past_events_timestamps")
+
+    late_event_ts = datetime.strptime(late_event_ts["timestamp"], TIMESTAMP_FORMAT)
+    for ts in late_events["late_events"]["hours"]:
+        current_event_ts = datetime.strptime(ts, TIMESTAMP_FORMAT)
+        if late_event_ts.microsecond == current_event_ts.microsecond:
+            difference_in_minutes = abs(current_event_ts - late_event_ts).seconds // 3600
+            assert TIME_UNIT["hours"]["min"] <= difference_in_minutes <= TIME_UNIT["hours"]["max"]
+
+
+def test_create_late_event_days(late_events, setup, tollways):
+    late_event_ts = create_late_event(
+        events_log=late_events, fake=setup.get("faker"), tollways=tollways, time_unit="days"
+    )
+
+    late_event_ts = datetime.strptime(late_event_ts["timestamp"], TIMESTAMP_FORMAT)
+    for ts in late_events["late_events"]["days"]:
+        current_event_ts = datetime.strptime(ts, TIMESTAMP_FORMAT)
+        if late_event_ts.microsecond == current_event_ts.microsecond:
+            difference_in_days = abs(current_event_ts - late_event_ts).days
+            assert TIME_UNIT["days"]["min"] <= difference_in_days <= TIME_UNIT["days"]["max"]
```

Comment:
assert in loop -> can you use parametrize here rather?

Reply:
also not sure about the if in the tests, ideally these should not be a need for those in tests, maybe we need to review the test data setup, is it too complex or not well-suited for the tests?

Reply:
It might be too complex but I'll see if I can put some of the setup in conftest along with utilizing parametrize

Reply:
Thanks, less urgent though, just wanted to share my doubts about flow control in test code.

Reply:
Maybe also a nice one to have ChatGPT help you with ...

Reply:
good call. I will. Have been using ChatGPT a lot more the last couple weeks.

Reply:
Yeah I still use it a lot, how is it going for you? Any surprises?

Reply:
it's going really well and feel like I'm getting better at using it. Starting to default more to let's ask ChatGPT then searching Google. I've subscribed to use GPT4 so feel somewhat more confident about the results. Still find myself asking the same question different ways just to ensure quality is mostly there. I've been using it with typing/type hints, refactoring, suggestions about different approaches, and different use cases on GCP for work.

Reply:
Awesome, glad to hear that! I actually just launched a poll here: https://twitter.com/bbelderbos/status/1750805206781071386

Reply:
And here: https://pybites.circle.so/c/off-topic-discussion-public/the-future-of-programming

Reply:
And yes GPT4 is way better, no only accuracy, also the amount of detail + depths it provides.

Lastly, here is an interesting view on ai tools as well: https://pybites.circle.so/c/share-your-wins-2deb20/browser-extension (towards the end)